lib/HLSL/HLSignatureLower.cpp

```diff
@@ -387,6 +392,48 @@ void HLSignatureLower::ProcessArgument(Function *func,
     pSE->SetOutputStream(streamIdx);
 }
 
+void HLSignatureLower::RemapObsoleteSemantic(DxilParameterAnnotation &paramInfo, const SigPoint *pSigPoint) {
+  DXASSERT(HLM.GetHLOptions().bBackCompatMode, "should be used only in back-compat mode");
+  DXASSERT(paramInfo.HasSemanticString(), "expected paramInfo with semantic");
+
+  llvm::StringRef semFullName = paramInfo.GetSemanticStringRef();
+  llvm::StringRef semName;
+  unsigned semIndex;
+  Semantic::DecomposeNameAndIndex(semFullName, &semName, &semIndex);
+
+  DXIL::SigPointKind sigPointKind = pSigPoint->GetKind();
+  if (sigPointKind == DXIL::SigPointKind::PSOut) {
+    if (semName.size() == 5) {
+      if (strnicmp(semName.data(), "COLOR", 5) == 0) {
+        RemapSemanticAndWarn(semName, semFullName, "SV_Target", paramInfo);
+      }
+      else if (strnicmp(semName.data(), "DEPTH", 5) == 0) {
+        RemapSemanticAndWarn(semName, semFullName, "SV_Depth", paramInfo);
+      }
+    }
+  }
+  else if ((sigPointKind == DXIL::SigPointKind::VSOut && semName.size() == 8 && strncmp(semName.data(), "POSITION", 8) == 0) ||
+           (sigPointKind == DXIL::SigPointKind::PSIn  && semName.size() == 4 && strncmp(semName.data(), "VPOS", 4) == 0)) {
```

Comment:
String compare is still case-sensitive for POSITION and VPOS.

Reply:
thanks!